seed/static/seed/js/controllers/data_quality_admin_controller.js

```diff
@@ -231,16 +276,46 @@ angular.module('BE.seed.controller.data_quality_admin', [])
         spinner_utility.show();
         data_quality_service.save_data_quality_rules($scope.org.org_id, rules).then(function (rules) {
           loadRules(rules);
-          $scope.rules_updated = true;
           modified_service.resetModified();
         }).then(function (data) {
           $scope.$emit('app_success', data);
         }).catch(function (data) {
           $scope.$emit('app_error', data);
         }).finally(function () {
+          $scope.rules_updated = true;
           spinner_utility.hide();
         });
       };
+      $scope.change_condition = function (rule) {
+        $scope.rules_updated = false;
+        $scope.defaults_restored = false;
+        $scope.rules_reset = false;
+        if (!rule.condition) rule.condition = null;
+        if (rule.condition === 'include' || rule.condition === 'exclude' && rule.data_type !== 'string') rule.data_type = 'string';
+        if (_.isMatch(rule, {condition: 'range', data_type: 'string'})) rule.data_type = null;
+        if (_.isMatch(rule, {condition: 'not_null', data_type: 'string'})) rule.text_match = null;
+
+        var condition = rule.condition;
+        if (condition === 'required') {
+          rule.required = false;
```

Comment:
Should we stop referring to and relying on the `required` and `not_null` attributes on the frontend? It looks like we're still referring to it in some capacity throught out the this controller.

This likely part of a larger discussion, but I assumed those particular Rule attributes would be removed from the database, model, etc. once we're fully converted over to using the `condition` values. I can imagine that removal happening down the road, and I could see an argument for making this refactor (of not using `required` and `not_null` attributes anymore) during that time.

Reply:
I believe we should remove not_null/required columns from database and model. And removing these 2 attributes is the intention. I agree that once the feature is stable we will no longer need these 2 columns for condition reference.

Reply:
Per our conversation, we're removing that stale checkbox-related test.